pkg/controller/nexus/resource/resources.go

```diff
@@ -71,21 +76,26 @@ func (r *nexusResourceManager) GetDeployedResources(nexus *v1alpha1.Nexus) (reso
 		resources, err = reader.ListAll(&v1.PersistentVolumeClaimList{}, &v1.ServiceList{}, &appsv1.DeploymentList{})
 	}
 
-	// ingress cannot be listed using utils, so we load the single one
-	ingressResType := reflect.TypeOf(v1beta1.Ingress{})
-	ingress, err := reader.Load(ingressResType, nexus.Name)
-	if err == nil {
-		resources[ingressResType] = []resource.KubernetesResource{ingress}
+	// Necessary to support OCP 3.x clusters
+	if available, err := kubernetes.IsIngressAvailable(r.discoveryClient); err != nil {
```

Comment:
We actually need this check so it doesn't crash on clusters that don't support Ingress. The way it was before, the  section in the `else if available` branch below would just be executed without checking whether or not Ingresses were available, which caused a panic IIRC.

If we insert this on the `else` above we're assuming that if Routes are not available then an Ingress is, which isn't necessarily the case with old clusters and will lead to a crash as well. I don't have a strong opinion about dropping support for OCP 3.11 (thus allowing us to put the section below on the `else` above and always assume Ingresses are available), though keeping it would make my testing life easier as I can't run CRC on my laptop. :laughing: 

But you're spot on about OCP only using routes here, the check if performed on the reconcile loop [here](https://github.com/m88i/nexus-operator/blob/4fb264e7674ac0e2ba38494c49978c91c57b6634/pkg/controller/nexus/nexus_controller.go#L219).

Reply:
Fine then!

Reply:
Alright, let's drop OCP 3.x. From now on we can always assume Ingresses are available. The king is dead, long live the king \o/